### PR TITLE
feat: Add `appmap-agent-config` command to bootstrap appmap.yml

### PR DESCRIPTION
**Pull request by dustinbyrne**
This command will be used during the AppMap manual install process to initialize an `appmap.yml`.

### CODE REVIEW

File: exe/appmap-agent-config [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# frozen_string_literal: true
+
+require 'optparse'
+require 'appmap'
+require 'appmap/command/agent_setup/config'
+
+@options = { config_file: AppMap::DEFAULT_CONFIG_FILE_PATH }
+
+OptionParser.new do |parser|
+  parser.banner = 'Usage: appmap-agent-init [options]'
```
Comment (kgilpin) [inline exe/appmap-agent-config:11]: Command name needs to be updated here

File: lib/appmap/command/agent_setup/config.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+module AppMap
+  module Command
+    module AgentSetup
+      ConfigStruct = Struct.new(:config_file)
+
+      class Config < ConfigStruct
+        def perform
+          if File.exist?(config_file)
+            puts "AppMap configuration file already exists at #{config_file}"
+            exit(1)
```
Comment (kgilpin) [inline lib/appmap/command/agent_setup/config.rb:15]: I feel like process exit should be handled in the `exe` package? So maybe just raise or return false.

File: exe/appmap-agent-config [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  description = "AppMap configuration file path (default: #{AppMap::DEFAULT_CONFIG_FILE_PATH})"
```
Comment (kgilpin) [inline exe/appmap-agent-config:13]: Probably want to update this to indicate that the file will be created by this command, that it won't be overwritten though. Maybe a `-f, --force` option to replace it? That would enable a developer to use it to "reset" the configuration.

Comment (dustinbyrne) [reply exe/appmap-agent-config:13]: Good idea. I've added the `--force`/`-f` option to overwrite any existing config.

File: lib/appmap/command/agent_setup/config.rb (hunk 2 above)
Comment (dustinbyrne) [reply lib/appmap/command/agent_setup/config.rb]: I've changed it to raise an error

**Review by kgilpin — APPROVED** [review]

**Pull request merged**